### PR TITLE
Upgrade to msw 1.0.x

### LINKED ISSUE #8340: Update msw to 1.0.X
Upgrade msw from 0.36.8 to 1.0.X. Will still not be able to upgrade to Node 18 sadly.

### PR DESCRIPTION
**Pull request by jpandersen87**
Fixes #8340 
This PR:
- updates msw from 0.36.8 to 1.0.X
- provides a new test utility that accepts matcher strings to suppress expected console warnings and errors via jest mockImplementation
- utility returns a function that can be called to restore console to original state
- uses said utility in various areas that tested error and warning handling
- fixes some errors that occurred in tests (these errors were not failure triggers for said tests)
- various tests were fixed due to behavior changes on hook tests from msw update
- changes to QueryClient instantiation to ensure a new client is created on every render to fix new errors likely related to poor isolation
- various hook tests that were bundled together were isolated into their own modules to fix new errors likely related to poor isolation

Hopefully with this update we'll be ready for a simple dependency bump when msw gets node 18 support.

### CODE REVIEW

**Comment by snesm** [comment]
Related to #8404

File: frontend-react/src/__mocks__/WatersMockServer.ts [hunk 1]
```diff
@@ -18,12 +18,12 @@ export enum WatersTestHeaderValue {
 const handlers = [
     rest.post(`${RS_API_URL}/api${WatersUrls.VALIDATE}`, (req, res, ctx) => {
         if (
-            req.headers["_headers"][WatersTestHeader.CLIENT] ===
+            req.headers.get(WatersTestHeader.CLIENT) ===
```
Comment (stephenkao) [inline frontend-react/src/__mocks__/WatersMockServer.ts:21]: 🙏

File: frontend-react/src/components/Admin/AdminReceiverDashboard.tsx [hunk 2]
```diff
@@ -547,11 +547,13 @@ function renderAllReceiverRows(props: {
                     </div>
                 </Grid>
                 <ScrollSyncPane enabled>
-                    <Grid row className={"horizontal-scroll"}>
-                        <Grid row className={"week-column"}>
-                            {perDayElements}
+                    <>
```
Comment (stephenkao) [inline frontend-react/src/components/Admin/AdminReceiverDashboard.tsx:550]: Out of curiosity, why is this needed?

File: frontend-react/src/components/Admin/EditSenderSettings.test.tsx [hunk 3]
```diff
@@ -33,8 +34,8 @@ jest.mock("react-router-dom", () => ({
     },
     useParams: () => {
         return {
-            orgName: "abbott",
```
Comment (stephenkao) [inline frontend-react/src/components/Admin/EditSenderSettings.test.tsx:36]: Welp

File: frontend-react/src/components/Admin/EditSenderSettings.test.tsx [hunk 4]
```diff
@@ -64,8 +65,15 @@ const testKeys = JSON.stringify([
 const testProcessingType = "sync";
 
 describe("EditSenderSettings", () => {
-    beforeAll(() => settingsServer.listen());
-    afterEach(() => settingsServer.resetHandlers());
+    beforeAll(() => {
+        settingsServer.listen();
+        settingsServer.use(
+            rest.get(
+                "https://test.prime.cdc.gov/api/settings/organizations/abbott/senders/user1234",
```
Comment (stephenkao) [inline frontend-react/src/components/Admin/EditSenderSettings.test.tsx:72]: I'm wondering if we should use the `REACT_APP_BACKEND_URL` environment variable for the base path here.  Also, I think it'd be good to make sure this is an invalid base path so we're not actually making calls to test.prime.cdc.gov since that's real.

Not an issue with Jest tests since they should be sandboxed, but I'm thinking we could adopt the pattern for all tests, including Cypress. 🤔

File: frontend-react/src/components/FileHandlers/FileHandlerMessaging.test.tsx [hunk 5]
```diff
@@ -63,6 +64,7 @@ describe("FileSuccessDisplay", () => {
 
 describe("FileErrorDisplay", () => {
     test("renders expected content", async () => {
+        const restore = conditionallySupressConsole("failure:");
```
Comment (stephenkao) [inline frontend-react/src/components/FileHandlers/FileHandlerMessaging.test.tsx:67]: Gonna add this to the upcoming tickets so we don't `console.error` on all validator errors 😬

File: frontend-react/src/components/Admin/AdminReceiverDashboard.tsx (hunk 2 above)
Comment (jpandersen87) [reply frontend-react/src/components/Admin/AdminReceiverDashboard.tsx:550]: A react ref error was being thrown that could only be resolved by giving the scroll panel a class-based component.

File: frontend-react/src/hooks/UseCreateFetch.ts [hunk 6]
```diff
@@ -45,11 +45,12 @@ function createTypeWrapperForAuthorizedFetch(
             headers,
         });
 
-        return axios(axiosConfig)
-            .then(({ data }) => data)
-            .catch((e: AxiosError) => {
-                throw new RSNetworkError<T>(e as AxiosError<T>);
-            });
+        try {
```
Comment (stephenkao) [inline frontend-react/src/hooks/UseCreateFetch.ts:48]: 🙏

File: frontend-react/src/hooks/UseValueSetActivation.test.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    beforeAll(() => lookupTableServer.listen());
+    afterEach(() => lookupTableServer.resetHandlers());
+    afterAll(() => lookupTableServer.close());
+
+    const renderWithQueryWrapper = () =>
+        renderHook(() => useValueSetActivation(), {
+            wrapper: QueryWrapper(),
+        });
+
+    test("returns trigger and loading indicator", async () => {
```
Comment (stephenkao) [inline frontend-react/src/hooks/UseValueSetActivation.test.ts:18]: Does this need to be `async`?

File: frontend-react/src/hooks/UseValueSets.ts [hunk 8]
```diff
@@ -92,10 +92,15 @@ export const useValueSetsMeta = (
     dataTableName: string = LookupTables.VALUE_SET
 ): ValueSetsMetaResponse => {
     const { authorizedFetch, rsUseQuery } = useAuthorizedFetch<LookupTable[]>();
+    const memoizedDataFetch = useCallback(
```
Comment (stephenkao) [inline frontend-react/src/hooks/UseValueSets.ts:95]: Why does this need to be memoized?  Even if the `authorizedFetch` reference changes, react-query should still prevent another request if the query key doesn't change, no?

File: frontend-react/src/hooks/filters/mocks/MockFilterManager.ts [hunk 9]
```diff
@@ -7,7 +7,7 @@ import { PageSettingsAction } from "../UsePages";
 
 const fakeDispatch = <T>(): Dispatch<T> => {
     return (_v: T) => {
-        console.log(_v);
+        // console.log(_v);
```
Comment (stephenkao) [inline frontend-react/src/hooks/filters/mocks/MockFilterManager.ts:10]: Eh?

File: frontend-react/src/network/QueryClients.ts [hunk 10]
```diff
@@ -12,7 +12,12 @@ export const appQueryClient = new QueryClient({
...
     },
 });
-export const testQueryClient = new QueryClient({
-    // to allow for faster testable failures
-    defaultOptions: { queries: { retry: false } },
-});
+export const getTestQueryClient = () =>
+    new QueryClient({
+        // to allow for faster testable failures
+        defaultOptions: {
```
Comment (stephenkao) [inline frontend-react/src/network/QueryClients.ts:18]: Since our approach is moving toward instantiating a new QueryClient for each test case, should we use [the same configuration as in production](https://github.com/CDCgov/prime-reportstream/pull/8349/files#diff-aa5131257f5b08b05099cac4d6c0fa3c367c4f16059043512369f72d668d6494R6-R11) so the behavior is the same?

File: frontend-react/src/utils/TestUtils.ts [hunk 11]
```diff
@@ -20,4 +20,35 @@ const mockEvent = (mock?: Partial<any>) => {
     };
 };
 
-export { mockEvent };
+function conditionallySupressConsole(...matchers: string[]) {
```
Comment (stephenkao) [inline frontend-react/src/utils/TestUtils.ts:23]: Thanks for adding this!  Super useful!

File: frontend-react/src/utils/TestUtils.ts [hunk 12]
```diff
@@ -20,4 +20,35 @@ const mockEvent = (mock?: Partial<any>) => {
...
+            }
+        });
+
+    return () => {
+        jestError.mockRestore();
+        jestWarn.mockRestore();
+    };
+}
+
+export { mockEvent, conditionallySupressConsole };
```
Comment (stephenkao) [inline frontend-react/src/utils/TestUtils.ts:54]: This is a stupid, pedantic, and useless point, but I feel like inline exports (`export function blahblahblah() { ... }`) are easier to think about than scrolling up/down in the file to make sure they're exported from a specific file.  Tiny, tiny subjective thing, though!

File: frontend-react/src/utils/TestUtils.ts (hunk 11 above)
Comment (stephenkao) [reply frontend-react/src/utils/TestUtils.ts:23]: That said, I think we should use caution when using this.  It looks like most/all of the warnings/errors in this pull request shouldn't be escalated to the test log, but I'm wary of overusing this just to create a silenced log.  Our first inclination should be to look into the why warnings/errors are getting printed out for a given test and silence them only if there's nothing actually fishy with said test.  We can also explicitly `expect` them if they're intentional, of course.

Not asking for any changes -- just something for the team to keep in mind!

**Review by stephenkao — APPROVED** [review]
Looks great!  Thanks for tackling this!

File: frontend-react/src/utils/TestUtils.ts (hunk 11 above)
Comment (jpandersen87) [reply frontend-react/src/utils/TestUtils.ts:23]: Very true. This is only to be used for scenarios where we are expecting console messages (error handling testing and such) to reduce noise to see real console messages that need addressing. I was debating on whether the return should be a tuple or object containing the spy objects so that it's possible to check the call counts and such.

Comment (stephenkao) [reply frontend-react/src/utils/TestUtils.ts:23]: No worries!  I think it's fine to keep this a separate function named as such so it gets attention called to it during code review.  Otherwise, we can manually set up spies when we're intending to catch warnings/errors.

One small, small thing, though -- I think `suppress` has two `p`s 😆

File: frontend-react/src/network/QueryClients.ts (hunk 10 above)
Comment (jpandersen87) [reply frontend-react/src/network/QueryClients.ts:18]: I ran into issues where turning on the suspend feature created errors when testing hooks in isolation.

File: frontend-react/src/hooks/filters/mocks/MockFilterManager.ts (hunk 9 above)
Comment (jpandersen87) [reply frontend-react/src/hooks/filters/mocks/MockFilterManager.ts:10]: The magic of main-merging

File: frontend-react/src/hooks/UseValueSets.ts (hunk 8 above)
Comment (jpandersen87) [reply frontend-react/src/hooks/UseValueSets.ts:95]: This was from earlier on when troubleshooting errors (value sets was a real trouble area) and I was throwing everything at the wall to see what would stick (aka stop erroring). I can revert if needed.

File: frontend-react/src/components/Admin/EditSenderSettings.test.tsx (hunk 4 above)
Comment (jpandersen87) [reply frontend-react/src/components/Admin/EditSenderSettings.test.tsx:72]: Funny enough, I'm pretty sure with this msw update preflight calls were succeeding to test.prime.cdc.gov which only stopped once I started isolating hook tests.

File: frontend-react/src/hooks/filters/mocks/MockFilterManager.ts (hunk 9 above)
Comment (jpandersen87) [reply frontend-react/src/hooks/filters/mocks/MockFilterManager.ts:10]: Woops misread the diff. I commented out the console log to reduce test noise. Should this log stay?

File: frontend-react/src/components/Admin/AdminReceiverDashboard.tsx (hunk 2 above)
Comment (stephenkao) [reply frontend-react/src/components/Admin/AdminReceiverDashboard.tsx:550]: Huh!  That's very odd...I wonder if it has something to do with [`childRef` not being used](https://github.com/okonet/react-scroll-sync/blob/master/src/ScrollSyncPane.js#L36) 🤔 

In any case, if a Fragment works for our cases (and it looks like it does), that's fine enough for me.

File: frontend-react/src/hooks/filters/mocks/MockFilterManager.ts (hunk 9 above)
Comment (stephenkao) [reply frontend-react/src/hooks/filters/mocks/MockFilterManager.ts:10]: Let's remove the log call, but I feel like each of these dispatches could be replaced with a spy instead.  Not a huge deal either way -- hoping we can get off of FilterManager and CursorManager soon.

File: frontend-react/.env.test [hunk 13]
```diff
@@ -3,8 +3,8 @@ REACT_APP_TITLE=CDC Prime ReportStream TEST
 REACT_APP_DESCRIPTION=ReportStream Dashboard Application
 REACT_APP_ICON=https://test.reportstream.cdc.gov/assets/cdc-logo.svg
 
-REACT_APP_BASE_URL=https://test.reportstream.gov
-REACT_APP_BACKEND_URL=https://test.prime.cdc.gov
+REACT_APP_BASE_URL=https://localhost
```
Comment (stephenkao) [inline frontend-react/.env.test:6]: 🙏

**Review by stephenkao — APPROVED** [review]
Looks great again! :)
